### PR TITLE
Add some documentation to the item class [CR]

### PR DESCRIPTION
**Pull request by BevapDin**
The main point of this is to document the function (e.g. for usage in Lua scripts).

There are some small code changes:
- Only one `item::color` function (without parameter), use `color_in_inventory` instead of `color(&g->u)`.
- `item::info` takes a vector reference and no `debug` parameter.
- Removes `item::getlight_emit_active`
- `item::clear` split into two (better named) functions.
- Moves content of `item::load_info` into `item::load_legacy`
- Moves `item::noise` into ranged.cpp and combines it with `make_gun_sound_effect` to be consistent. (The list of silent ammo types was repeated twice.)
- Makes some item members private.

### CODE REVIEW

**Comment by kevingranade** [comment]
Looks like some heavy-duty cleanup, I don't see anything to object to.

File: doc/JSON_FLAGS.md [hunk 1]
```diff
@@ -761,6 +764,9 @@ Melee flags are fully compatible with tool flags, and vice versa.
 - ```NO_DROP``` An item with this flag should never actually be dropped. Used internally to signal that an item was created, but that it is unwanted. Needless to say, don't use this in an item definition.
 - ```WET``` Item is wet and will slowly dry off (e.g. towel).
 - ```MC_MOBILE```, ```MC_RANDOM_STUFF```, ```MC_SCIENCE_STUFF```, ```MC_USED```, ```MC_HAS_DATA``` Memory card related flags, see `iuse.cpp`
+- ```HAS_RECIPE``` Used by the E-Ink tablet to indicates it's currently showing a recipe.
+- ```RADIO_MODABLE``` Indicates the item can be made into a radio-activated item.
+- ```RADIO_MO``` The item has been made into a radio-activated item.
```
Comment (Coolthulhu) [inline doc/JSON_FLAGS.md]: Tiny typo: `RADIO_MOD`

File: src/item.h [hunk 2]
```diff
@@ -221,17 +238,49 @@ class item : public JsonSerializer, public JsonDeserializer
...
+    /**
+     * @name Melee
+     *
+     * The functions here assume the item is used in melee, even if's a gun or not a weapon at
+     * all. Because the functions apply to all types of items, several of the is_* functions here
+     * may return true for the same item. This only indicates that it can be used in various ways.
+     */
+    /*@{*/
+    /**
+     * Number of moves (@ref Creature::moves) that a single attack take.
```
Comment (Coolthulhu) [inline src/item.h]: Should probably be "base number of moves". Attack time is heavily dependent on attacker.

File: src/item.h [hunk 3]
```diff
@@ -221,17 +238,49 @@ class item : public JsonSerializer, public JsonDeserializer
...
+     */
+    int damage_cut() const;
+    /**
+     * Whether the character needs both hands to wield this item.
+     */
+    // TODO: make a reference. Make a const reference.
+    bool is_two_handed(player *u);
+    /** The weapon is considered a suitable melee weapon. */
+    bool is_weap() const;
+    /** The weapon inflicts at least a certain minimal basing damage. */
```
Comment (Coolthulhu) [inline src/item.h]: Why not just explicitly say how much? This number probably won't change soon.

File: src/ranged.cpp [hunk 4]
```diff
@@ -1359,16 +1399,17 @@ void make_gun_sound_effect(player &p, bool burst, item *weapon)
     }
 
     if( ammo_used == "40mm") {
-        sounds::sound(p.pos(), 8, _("Thunk!"), false, "fire_gun", weapon->typeId());
+        gunsound = _("Thunk!");
```
Comment (Coolthulhu) [inline src/ranged.cpp]: Not exactly related to this PR, but those should probably be jsonized.

File: src/item.h (hunk 3 above)
Comment (BevapDin) [inline src/item.h]: > Why not just explicitly say how much? This number probably won't change soon.

The number should be of no interest to the read. In fact, it's an implementation detail, the reader of the documentation should not depend on that behavior.

But the wording is bit unclear (and in its current form not much better than showing the number). I changed it to show what the functions intention is.

**Pull request merged**